.github/workflows/ci.yml

```diff
@@ -60,15 +60,21 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Python
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+        with:
+          version: "0.4.29"
+          enable-cache: true
+          cache-dependency-glob: "./projects/pgai/uv.lock"
+
+      - name: "Set up Python"
         uses: actions/setup-python@v5
         with:
-          python-version: "3.12"
-          cache: "pip" # caching pip dependencies
+          python-version-file: "./projects/pgai/.python-version"
```

Comment:
We have the minimum version of python set to 3.10 but are using 3.12 in the docker image. I feel like we might want to use a test matrix here to ensure it works with 3.11, 3.12, 3.13?

For development I think it makes sense to use the minimum python version so that we don't accidentally make use of newer features than we wanted to support.

Reply:
I think the test matrix is a good idea. Agree on using the python version file with 3.10 for development.

Reply:
The timescaledb ha image uses Python 3.10 which is therefore what we have to use for the extension. I believe the pgai library can use whatever we wish since it is separate.

Reply:
yes, but if as a user your project is e.g. not yet python 3.12 you will run into compatibility issues when you try to install the package.

Tbf right now there is little need to install pgai into an existing project but if we add more python helpers, this might become important and I would suggest that we roughly follow the [release and support cadence of python](https://devguide.python.org/versions/) for this.

Reply:
I'll do test matrix in a separate PR.